### ISSUE TITLE
Final output has many gene annotations in repetitive regions

### ISSUE THREAD

**Issue by ZuyaoLiu** [sagnikbanerjee15/Finder#33]
@sagnikbanerjee15 

I found that in the final output, many genes are located in repeat regions.
It seems the soft masking of the genome doesn't work well.

Got any idea?

Thanks

**Comment by sagnikbanerjee15** [comment]
Hi @ZuyaoLiu,

Thank you for pointing it out. `finder` should flag those genes present in the repeat regions and not put those in the high confidence group. Which file are you looking at? Also, could you please send me the genome and the final annotations you received?

Thank you.

**Comment by ZuyaoLiu** [comment]
@sagnikbanerjee15 
I have been trying to debug the script removeSpuriousTranscriptsBasedOnCDS.py. For me, everything looks fine so far. But the filtration on repeat doesn't work at all. I tried to change the cut-off from 0.25 to 0.9, which I guess should make some difference. But the output files are still the same as before.
Therefore I get a gff with 80,000 genes. I used bedtools intersect to remove those overlapped with my repeat_region.gff3, the gene number dropped to around 26,000, which makes sense

**Comment by sagnikbanerjee15** [comment]
Hmm, Not sure why it would not work. Did you remove the files and try again? `finder` by default will skip files that are already present to prevent taking more time.

Thank you.

**Comment by ZuyaoLiu** [comment]
> Hmm, Not sure why it would not work. Did you remove the files and try again? `finder` by default will skip files that are already present to prevent taking more time.
> 
> Thank you.

@sagnikbanerjee15 
Now I remove everything and restart the pipeline. I will tell you what I got eventually. Let's see.

**Comment by ZuyaoLiu** [comment]
@sagnikbanerjee15 
I rerun the pipeline, and the genes in repetitive regions seem still to be not removed.
Here is the file size. The file size with total genes that are found based on RNA assembly is 126MB. After filtration, the file size is 124MB, which doesn't make sense since a lot of genes are located in repetitive regions. Also I tried to increase the cutoff but it didn't change the result.
![微信截图_20211108110258](https://user-images.githubusercontent.com/49354864/140722563-8381f3f5-61b5-4803-880c-049f50b67f6e.png)

Could you please check it?
Thanks.

Zuyao

**Comment by sagnikbanerjee15** [comment]
Hi @ZuyaoLiu,

Thank you for reporting this. Could you please check the actual number of genes and transcripts in each of those files? Some transcripts could be monoexonic and would not occupy much space on the file. Hence, their removal may not substantially reduce the file size.

Thank you.

**Comment by ZuyaoLiu** [comment]
Hi @sagnikbanerjee15 

Here are the results.
For combined_with_CDS.gtf: 84761
For combined_with_CDS_high_conf.gtf: 83365
For combined_with_CDS_low.gtf: 3986
I sorted the attributes column based on the gene_id and removed the redundancy, excluding the alternative splices.

That seems not reliable, since I test with Braker-only prediction pipeline as well as maker3. Both programs predict about 24000 genes.
And within the pipeline of Finder the Braker gives around 26000 genes, which is normal, therefore I guess the issue is with the filtration step.

Zuyao

**Comment by sagnikbanerjee15** [comment]
Yes you are right. This does not look like what I expected. Could you please send me the genome file along with the gtf files? I need to explore further. Also, please send me the metadata file alongwith the protein file if you had used one.

Thank you.

**Comment by ZuyaoLiu** [comment]
Hi @sagnikbanerjee15 
I upload the files to Google drive. Here is the link: https://drive.google.com/file/d/1z37gEJSP1mLMIQrE5LlB5WITLFhkqeBZ/view?usp=sharing
For testing, I upload one chromosome fasta file , meta.csv, gtf files in the screensot and the proteins I used (mainly from uniprot and a closely related species).

If you need anything else, please let me know. I'm glad to help.
Zuyao
